### PR TITLE
feat(api): check if scalingFactor is bigger than 0

### PR DESCRIPTION
**Pull request by 317787106**
**What does this PR do?**

- rollback the changes in https://github.com/tronprotocol/java-tron/pull/5688
- throw exception if scalingFactor <= 0 in checkPublicAmount.

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: framework/src/main/java/org/tron/core/Wallet.java [hunk 1]
```diff
@@ -4127,12 +4129,11 @@ public byte[] getShieldedContractScalingFactor(byte[] contractAddress)
     TransactionExtention.Builder trxExtBuilder = TransactionExtention.newBuilder();
     Return.Builder retBuilder = Return.newBuilder();
     TransactionExtention trxExt;
-    Transaction trx;
 
     try {
       TransactionCapsule trxCap = createTransactionCapsule(trigger,
           ContractType.TriggerSmartContract);
-      trx = triggerConstantContract(trigger, trxCap, trxExtBuilder, retBuilder);
+      Transaction trx = triggerConstantContract(trigger, trxCap, trxExtBuilder, retBuilder);
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/core/Wallet.java:4136]: Doesn't this need to be handled?

Comment (317787106) [reply framework/src/main/java/org/tron/core/Wallet.java:4136]: This part is rollback to 4.7.3.

Comment (halibobo1205) [reply framework/src/main/java/org/tron/core/Wallet.java:4136]: What is the reason for this part of the direct rollback ? I see the previous analysis which is also wrong

Comment (317787106) [reply framework/src/main/java/org/tron/core/Wallet.java:4136]: Only this part is added: 
```
if (scalingFactor.compareTo(BigInteger.ZERO) <= 0) {
      throw new ContractValidateException("scalingFactor must be positive");
}
```

Other part is still same as before. Check the condition outside this method is not only easy to understand and but also include all conditions.

Comment (317787106) [reply framework/src/main/java/org/tron/core/Wallet.java:4136]: @halibobo1205 what's wrong with previous analysis? This optimization is to coverage this additional case: contract and method `scalingFactor` are both exist, but it really return 0.

**Review by jwrct — APPROVED** [review]

**Review by halibobo1205 — APPROVED** [review]

**Pull request merged**